pkgs/applications/misc/logseq/default.nix

```
@@ -29,6 +28,12 @@ stdenv.mkDerivation rec {
     cp -a ${appimageContents}/{locales,resources} $out/share/${pname}
     cp -a ${appimageContents}/Logseq.desktop $out/share/applications/${pname}.desktop
 
+    # remove the `git` in `dugite` because we want the `git` in `nixpkgs`
+    chmod +w -R $out/share/${pname}/resources/app/node_modules/dugite/git
+    chmod +w $out/share/${pname}/resources/app/node_modules/dugite
+    rm -rf $out/share/${pname}/resources/app/node_modules/dugite/git
+    chmod -w $out/share/${pname}/resources/app/node_modules/dugite
```

I think it's not necessary to change the permissions:
```suggestion
    rm -rf $out/share/${pname}/resources/app/node_modules/dugite/git
```
besides that, LGTM.

---

It seems that these files are extracted from appimage and with permission `r-xr-xr-x`. So it's necessary to set the write permission to remove the `git` directory.

I'm not familiar with `nix` and I get the permission info as following:

```
$ ls -l $out/share/logseq/resources/app/node_modules
total 0
dr-xr-xr-x 1 nixbld1 nixbld   0 Jan  1  1970 '@develar'/
dr-xr-xr-x 1 nixbld1 nixbld   6 Jan  1  1970 '@electron'/
dr-xr-xr-x 1 nixbld1 nixbld   0 Jan  1  1970 '@electron-forge'/
......
dr-xr-xr-x 1 nixbld1 nixbld 230 Jan  1  1970  dugite/
......
```

---

What they do is not exactly the same.

`chmod +w -R $out/share/${pname}/resources/app/node_modules/dugite`
will recurse into other dirs besides `git`.

But maybe it's not a big problem... I don't know